src/common/logger.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"log/slog"
+	"os"
+)
+
+// Default logger for the entire system (Any call from log will immediately be written with a JSON handler)
```

Comment:
You have some more details comments about logging and locking in the PR itself.  I think we should do most of the explanation in the actual comments.  Please include links/quotes to documentation where appropriate.